hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -160,9 +169,11 @@ public Map<String, Object> generateMetadataForRecord(
    * @param schema The Avro schema of the record.
    * @return A mapping containing the metadata.
    */
-  public Map<String, Object> generateMetadataForRecord(T record, Schema schema) {
+  public Map<String, Object> generateMetadataForRecord(T record, Schema schema, boolean isPartial) {
     Map<String, Object> meta = new HashMap<>();
     meta.put(INTERNAL_META_RECORD_KEY, getRecordKey(record, schema));
+    meta.put(INTERNAL_META_SCHEMA, schema);
+    meta.put(INTERNAL_META_IS_PARTIAL, isPartial);
```

Comment:
I'm wondering whether we can represent the metadata as a POJO to make the interface more explicit and clear.

Reply:
We can.  We can take it up in a separate PR.